packages/ai/core/generate-object/output-strategy.ts

```diff
@@ -335,11 +335,41 @@ const enumOutputStrategy = <ENUM extends string>(
           };
     },
 
-    validatePartialResult() {
-      // no streaming in enum mode
-      throw new UnsupportedFunctionalityError({
-        functionality: 'partial results in enum mode',
-      });
+    async validatePartialResult({ value, textDelta }) {
+      if (!isJSONObject(value) || typeof value.result !== 'string') {
+        return {
+          success: false,
+          error: new TypeValidationError({
+            value,
+            cause:
+              'value must be an object that contains a string in the "result" property.',
+          }),
+        };
+      }
+
+      const result = value.result as string;
+      const possibleEnumValues = enumValues.filter(enumValue =>
+        enumValue.startsWith(result),
+      );
+
+      if (value.result.length === 0 || possibleEnumValues.length === 0) {
+        return {
+          success: false,
+          error: new TypeValidationError({
+            value,
+            cause: 'value must be a string in the enum',
+          }),
+        };
+      }
+
+      return {
+        success: true,
+        value: {
+          partial:
+            possibleEnumValues.length > 1 ? result : possibleEnumValues[0],
+          textDelta,
+        },
+      };
```

Comment:
this can lead to issues where the wrong enum value is selected as a partial result, leading to changing enum values. i'd prefer a solution where that cannot happen (either by only returning fully matched enum values, or (more complicated) by returning enum values for which only one value is possible per prefix)

Reply:
yep good call out - I added tests to test for these cases. In this case I check for `possibleEnumValues.length > 1` which means there's more than one match, in that case i return just the partial value. If there's only one possible enum per prefix, i return the full value preemptively